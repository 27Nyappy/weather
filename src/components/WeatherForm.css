.WeatherForm {
  @media only screen and (max-width: 430px) {
    padding: 20px;
    width: 80%;

    section {

      div[id=zip] {
        display: flex;
        align-items: center;
      }

      div[id=days] {
        display: flex;
        align-items: center;
        justify-content: space-between;

        div[id=selections] {
          width: 50%;

          label {
            width: 60px;
          }
        }
      }
    }
  }

  @media screen and (min-width: 431px) and (max-width: 600px) {
    padding: 15px 30px;
    width: 80%;

    section {

      div[id=zip] {
        display: flex;
        align-items: center;
        justify-content: space-between;
      }

      div[id=days] {
        display: flex;
        align-items: center;
        justify-content: space-between;

        div[id=selections] {
          width: 50%;

          label {
            width: 65px;
          }
        }
      }
    }
  }

  @media screen and (min-width: 601px) and (max-width: 768px) {
    padding: 15px 30px;
    width: 80%;

    section {
    
      div[id=zip] {
        display: flex;
        align-items: center;
        justify-content: space-between;
      }

      div[id=days] {
        display: flex;
        align-items: center;
        justify-content: space-between;

        div[id=selections] {
          width: 35%;

          label {
            width: 65px;
          }
        }
      }
    }
  }

  @media screen and (min-width: 769px) and (max-width: 834px) {
    padding: 15px 50px 15px 30px;
    width: 75%;

    section {
      display: flex;
      flex-direction: row;
      justify-content: space-between;

      div[id=zip] {
        display: flex;
        flex-direction: column;
        align-items: center;
        width: 45%;
      }

      div[id=days] {
        display: flex;
        flex-direction: column;
        align-items: center;
        width: 45%;

        div[id=selections] {
          width: 90%;
  
          label {
            width: 65px;
          }
        }
      }
    }
  }

  @media screen and (min-width: 835px) {
    padding: 15px 50px 15px 30px;
    width: 70%;

    section {
      display: flex;
      flex-direction: row;
      justify-content: space-between;

      div[id=zip] {
        display: flex;
        flex-direction: column;
        align-items: center;
        width: 45%;
      }

      div[id=days] {
        display: flex;
        flex-direction: column;
        align-items: center;
        width: 45%;

        div[id=selections] {
          width: 100%;
  
          label {
            width: 120px;
          }
        }
      }
    }
  }

  display: flex;
  flex-direction: column;
  max-width: 600px;
  margin: 20px auto;
  background-color: #76dbd1;
  box-shadow: 0 0 3px #203f3c95;
  color: #555151;

  h2 {
    font-size: 13pt;
  }

  input[type=number] {
    margin: 10px 0;
    width: 200px;
    font-size: 11pt;
  }

  input[type=radio] {
    display: none;
  }

  input[type=radio]:checked + label {
    color: #0c0a0a;
    border-bottom: medium solid #026759;
  }

  div[id=selections] {
    display: flex;
    flex-direction: row;
    justify-content: space-between;
  }

  div {
    
    label {
      margin: 10px 0;
      border-bottom: medium solid #ecf4f3;
      font-size: 11pt;
      transition: 0.5s ease;
    }

    label:hover {
      cursor: pointer;
    }
  }

  input[type=submit] {
    box-sizing: content-box;
    background-color: #57a99a;
    box-shadow: 0 0 3px #57a99a;
    color: #ecf4f3;
    border: none;
    width: 120px;
    margin: 5px auto 0 auto;
    transition: 0.2s ease-in-out;
    font-family: 'Roboto', sans-serif;
    font-weight: 300;
    font-size: 12pt;
  }

  input[type=submit]:hover {
    cursor: pointer;
    transform: scale(1.1);
    box-shadow: rgba(128, 128, 128, 0.616);
    color: #57a99a;
    background-color: #ecf4f3;
  }
}