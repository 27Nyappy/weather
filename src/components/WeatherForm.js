import React from 'react';
import PropTypes from 'prop-types';
import styles from './WeatherForm.css';

const WeatherForm = ({ handleSubmit, handleZipChange, handleDayChange }) => {
  return (
    <form onSubmit={handleSubmit} className={styles.WeatherForm}>
      <section>
        <div id="zip">
          <h2>Please Enter Your Zip Code</h2>
          <input type="number" max="99950" min="00501" onChange={handleZipChange} placeholder="ZIP" required />
        </div>
        <div id="days">
          <h2>Select Time</h2>
          <div id="selections">
            <input type="radio" name="day-options" id="current-weather" onChange={handleDayChange} value="weather" required />
            <label htmlFor="current-weather">Current Weather</label>
            <input type="radio" name="day-options" id="five-day-weather" onChange={handleDayChange} value="forecast" required />
            <label htmlFor="five-day-weather">Five Day Forecast</label>
          </div>
        </div>
      </section>
      <input type="submit" value="Get Weather" />
    </form>
  );
};

WeatherForm.propTypes = {
  handleSubmit: PropTypes.func.isRequired,
  handleDayChange: PropTypes.func.isRequired,
  handleZipChange: PropTypes.func.isRequired
};

export default WeatherForm;