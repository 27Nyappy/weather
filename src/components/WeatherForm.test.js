import React from 'react';
import { shallow } from 'enzyme';
import WeatherForm from './WeatherForm';

describe('Weather Form Component', () => {
  it('Renders Weather Form', () => {
    const wrapper = shallow(<WeatherForm handleSubmit={() => true} handleZipChange={() => true} handleDayChange={() => true} />);
    expect(wrapper).toMatchSnapshot();
  });
});