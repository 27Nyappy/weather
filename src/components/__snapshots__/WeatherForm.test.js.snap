// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Weather Form Component Renders Weather Form 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    max="99950"
    min="00501"
    onChange={[Function]}
    required={true}
    type="number"
  />
  <input
    id="current-weather"
    name="day-options"
    onChange={[Function]}
    required={true}
    type="radio"
    value="weather"
  />
  <label
    htmlFor="current-weather"
  >
    Current Weather
  </label>
  <input
    id="five-day-weather"
    name="day-options"
    onChange={[Function]}
    required={true}
    type="radio"
    value="forecast"
  />
  <label
    htmlFor="five-day-weather"
  >
    Five Day Forecast
  </label>
  <input
    type="submit"
    value="Get Weather"
  />
</form>
`;
